src/utils/getOrganisationDetails.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+const getOrganisationDetails = (
+  data: Record<string, any>[],
+  isDetails: boolean = false,
+): { organisationValue: string } => {
+  if (Array.isArray(data) && data.length > 0) {
+    const rolesOrder: string[] = ['custodian', 'pointOfContact', 'originator', 'distributor', 'owner'];
+
+    const getOrganisation = (role: string): string => {
+      const obj = data.find((item: Record<string, any>) => item.role === role);
+      return obj ? obj?.organisationObject?.default ?? '' : '';
+    };
+
+    if (isDetails) {
+      for (const role of rolesOrder) {
```

Comment:
as per the AC, we need data for a exact role set and not this way. has this been confirmed with Bill?

Reply:
We discussed this on Thursday. Bill also mentioned during the call that the 'Custodian' is mandatory for MEDIN, but we can check for other roles for other sources. Bill has sent me the message and attached a screenshot of the roles order we need to check. I asked Bill to update it in the AC and also I commented on the user story about this on Thursday.

image.png

Reply:
Now the heirarchy for the roles has been updated on the ticket.